text/0000-confidential-token-ids.md

```diff
@@ -0,0 +1,270 @@
+
+    ```
+    blinding_factor = Blake2B(blinding_tag | shared_secret)
+    commitment = H_i * Scalar::from(amount) + G * blinding_factor
+    ```
+
+    - Note, the group generators for Token Types are computed using hashing to a curve so that they are "orthogonal" for computable adversaries, meaning there is no discernable, nontrivial linear relationship between base points. See our [transaction core group generators reference](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/transaction/core/src/ring_signature/mod.rs#L29). 
+    - Note also, the blinding factor base point does not need to be unique per TokenType, so we use the same blinding factor base point for all TokenTypes.
+    - Note finally, the Proof of Opening requires all commitments in a transaction were generated using the same group generator, base point `H_i`. If the commitments in input and/or output TXOs in a transaction are constructed with different group generators, and the sender is able to unwrap the sum as specified in the protocol below, it implies that the sender knows a nontrivial linear relationship among the `H_i`, which contradicts the assumption that the `H_i` are orthogonal. 
+
+The non-interactive zero knowledge Proof of Opening protocol is the following: (See [Zero Knowledge Proofs and Commitment Schemes, Page 27](https://www.cs.purdue.edu/homes/ninghui/courses/555_Spring12/handouts/555_Spring12_topic23.pdf))
+
+
+1. The prover establishes a value, `d`, calculated by hashing their secret values from the commitment (the `token_id` and the `blinding_factor`). Implementation note, we may use [Merlin](https://docs.rs/merlin/latest/merlin/) rather than Blake2B.
+
+    ```
+    y = Blake2B(proof_tag_t | token_id)
+    s = Blake2B(proof_tag_b | blinding_factor)
+    d = G * y + H * s
+    ```
+
+2. The non-interactive verifier chooses challenge, `e`, by hashing the commitment and the value `d` from the previous step.
+
+    ```
+    e = Blake2B(proof_tag_e | commitment | d)
+    ```
+
+3. The prover computes the following, and includes these values in the Proof of Opening:
+
+    ```
+    u = y + e * token_id
+    v = s + e * blinding_factor
+    ```
+
+4. The verifier of the Proof of Opening must verify:
+
+    ```
+    G * u + H * v = d + commitment * e
+    ```
+
+    - This property follows a similar property of the [Schnorr Proof of Knowledge protocol](https://en.wikipedia.org/wiki/Proof_of_knowledge#Schnorr_protocol)
+
+    
+This protocol is non-interactive because the prover anticipates the verifier's `e` value, so requires no interaction from the verifier.
+
+Thus, the Proof of Knowledge could be represented, for example, with:
+
+```
+struct ProofOfKnowledge {
+
+    /// Prover value established from hashes of commitment and blinding factor secrets
+    CompressedRistretto d,
```

Comment:
Capital letters for curve points make it much easier to disambiguate curve points from curve scalars.